problem/0000-ci-health-metrics.md

```diff
@@ -0,0 +1,81 @@
+- Contribution Name: ci-health-metrics
+- Implementation Owner: @alpeb, @kleimkuhler
+- Start Date: 2020-03-16
+- Target Date: 2020-04-03
+- RFC PR: [linkerd/rfc#11](https://github.com/linkerd/rfc/pull/11)
+- Linkerd Issue:
+  [linkerd/linkerd2#4176](https://github.com/linkerd/linkerd2/issues/4176)
+- Reviewers: @grampelberg
+
+# Summary
+
+[summary]: #summary
+
+This RFC proposes a service that would monitor the results from each CI run. It
+should allow us to catch consistently flaky tests and have an overview of the CI
+in terms of success rates and time performance.
+
+# Problem Statement (Step 1)
+
+[problem-statement]: #problem-statement
+
+CI jobs inevitably fail occasionally, either because of real problems with the
+code, or because of tests flakiness. We'd like to be able to be informed about
+what tests are being persistently flaky, without having to dig into the CI run
+results themselves. This will allow us to properly prioritize work to fix those
+failures.
+
+Besides identifying the most common failures, by surfacing error metrics from the
+CI jobs we'll also have at our disposal an overview of CI's performance, in
+terms of things like success rates and time taken by the jobs. This information
+can be aggregated at the global, workflow and jobs levels.
+
+## Phase 1
+
+CI errors are surfaced as annotations in Github's Workflows and Checks APIs.
```

Comment:
I hope I'm not being too nitpicky here but this all sounds like solution-space stuff to me, rather than problem-statement. If we're splitting the problem into two chunks, what are the problems that are solved in the first step? Parsing output of scripts is an implementation detail of solving what problem?

Reply:
Phase 1 would solve the problem that test results aren't currently being exposed for consumption anywhere besides the Github UI.
Once data is exposed, Phase 2 would solve the problem of how to consume and present that data.
Does that make sense, or am I missing something else?

Reply:
Yep. I think this section should enumerate the problems, and the design proposal below should discuss workflows, checks, etc.